pkgs/build-support/setup-hooks/make-wrapper.sh

```diff
@@ -43,19 +45,20 @@ makeWrapper() {
                     echo "export $varName=$value\${$varName:+$separator}\$$varName" >> $wrapper
                 fi
             fi
-        fi
+            ;;
 
-        if test "$p" = "--suffix-each"; then
+          "--suffix-each")
             varName=${params[$((n + 1))]}
             separator=${params[$((n + 2))]}
             values=${params[$((n + 3))]}
             n=$((n + 3))
             for value in $values; do
                 echo "export $varName=\$$varName\${$varName:+$separator}$value" >> $wrapper
             done
-        fi
+            ;;
 
-        if test "$p" = "--suffix-contents" -o "$p" = "--prefix-contents"; then
+          "--suffix-contents") ;& # fallthrough
+          "--prefix-contents")
```

Comment:
not ideal but instead of using the fallthrough how about using `--*-contents` as the pattern? the test case below could be expanded to handle non-matching keys.